`website`: Add quality checks.

We need to add all the following to `website/package.json` wireit tasks:

- [x] `lint`
- [x] `lint:fix`
- [x] `prettier`
- [x] `prettier:fix`
- [ ] `types:check` (Note -- this is blocked by #1201 )

Then:

- [ ] Integrate all those checks in root aggregate tasks in `package.json`
- [ ] Remove `paths-ignore` in main CI workflow so we do all docs code checking there
- However, let's keep the `build` task in `website` out of both wireit and CI and leave it to just the docs workflow.

alrighty, so now I'm getting this error for `types:check` even on the `main` branch:

```
pnpm run types:check

> spectacle-monorepo@ types:check /Users/paulmarsicovetere/Documents/code/FormidableLabs/main_branches_only/spectacle
> wireit

🏃 [packages/create-spectacle:types:check] Running command "nps types:check -- -- --p tsconfig.typecheck.json"
🏃 [packages/spectacle:types:check] Running command "nps types:check"
🏃 [examples/typescript:types:check] Running command "nps types:check"
nps is executing `types:check` : tsc --noEmit
nps is executing `types:check` : tsc --noEmit --p tsconfig.typecheck.json
nps is executing `types:check` : tsc --noEmit
index.tsx(11,3): error TS2305: Module '"spectacle"' has no exported member 'AnimatedProgress'.
index.tsx(23,3): error TS2305: Module '"spectacle"' has no exported member 'SlideLayout'.
The script called "types:check" which runs "tsc --noEmit" failed with exit code 2 https://github.com/sezna/nps/blob/master/other/ERRORS_AND_WARNINGS.md#failed-with-exit-code
✅ [packages/create-spectacle:types:check] Executed successfully
✅ [packages/spectacle:types:check] Executed successfully
❌ [examples/typescript:types:check] Failed with exit status 2
 ELIFECYCLE  Command failed with exit code 1.
```

Once this is resolved, I think I should be good to PR for the `lint` and `prettier` changes 👍

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -8,13 +8,9 @@ on:
   push:
     branches:
       - main
-    paths-ignore:
-      - "docs/**"
   pull_request:
     branches:
       - main
-    paths-ignore:
-      - "docs/**"
 
 jobs:
   build:

--- a/package-scripts.js
+++ b/package-scripts.js
@@ -41,6 +41,10 @@ module.exports = {
       'nps prettier:base -- -- --config ../../.prettierrc --ignore-path ../../.prettierignore',
     'prettier:pkg:fix':
       'nps prettier:base:fix -- -- --config ../../.prettierrc --ignore-path ../../.prettierignore',
+    'prettier:website':
+      'nps prettier:base -- -- --config ../.prettierrc --ignore-path ../.prettierignore',
+    'prettier:website:fix':
+      'nps prettier:base:fix -- -- --config ../.prettierrc --ignore-path ../.prettierignore',
 
     // - Lint
     'lint:base': 'eslint --cache --color',

--- a/package.json
+++ b/package.json
@@ -166,6 +166,7 @@
       "dependencies": [
         "./packages/spectacle:lint",
         "./packages/create-spectacle:lint",
+        "./website:lint",
         "./examples/js:lint",
         "./examples/md:lint",
         "./examples/one-page:lint",
@@ -176,6 +177,7 @@
       "dependencies": [
         "./packages/spectacle:lint:fix",
         "./packages/create-spectacle:lint:fix",
+        "./website:lint:fix",
         "./examples/js:lint:fix",
         "./examples/md:lint:fix",
         "./examples/one-page:lint:fix",
@@ -224,6 +226,7 @@
       "dependencies": [
         "./packages/spectacle:prettier",
         "./packages/create-spectacle:prettier",
+        "./website:prettier",
         "./examples/js:prettier",
         "./examples/md:prettier",
         "./examples/one-page:prettier",
@@ -234,6 +237,7 @@
       "dependencies": [
         "./packages/spectacle:prettier:fix",
         "./packages/create-spectacle:prettier:fix",
+        "./website:prettier:fix",
         "./examples/js:prettier:fix",
         "./examples/md:prettier:fix",
         "./examples/one-page:prettier:fix",

--- a/website/docusaurus.config.js
+++ b/website/docusaurus.config.js
@@ -65,7 +65,7 @@ async function createConfig() {
     ],
 
     themeConfig:
-    /** @type {import("@docusaurus/preset-classic").ThemeConfig} */
+      /** @type {import("@docusaurus/preset-classic").ThemeConfig} */
       ({
         navbar: {
           title: 'SPECTACLE',

--- a/website/package.json
+++ b/website/package.json
@@ -49,8 +49,8 @@
     "lint": {
       "command": "nps lint:pkg -- -- \"*.js\" src",
       "files": [
-        "../../.eslintignore",
-        "../../.eslintrc",
+        "../.eslintignore",
+        "../.eslintrc",
         "*.js",
         "src/**"
       ],
@@ -73,7 +73,7 @@
       ]
     },
     "prettier": {
-      "command": "nps prettier:website -- -- src",
+      "command": "nps prettier:website -- -- \"*.js\" src",
       "files": [
         "../.prettierignore",
         "../.prettierrc",

--- a/website/package.json
+++ b/website/package.json
@@ -5,18 +5,6 @@
   "engines": {
     "node": "<18"
   },
-  "scripts": {
-    "docusaurus": "docusaurus",
-    "start": "docusaurus start",
-    "build": "docusaurus build --out-dir build/open-source/spectacle",
-    "clear": "docusaurus clear",
-    "serve": "formideploy serve",
-    "deploy:stage": "formideploy deploy --staging",
-    "deploy:prod": "formideploy deploy --production",
-    "write-translations": "docusaurus write-translations",
-    "write-heading-ids": "docusaurus write-heading-ids",
-    "typecheck": "tsc"
-  },
   "devDependencies": {
     "@docusaurus/core": "^2.0.1",
     "@docusaurus/module-type-aliases": "^2.0.1",
@@ -40,6 +28,76 @@
     "sass-loader": "^13.0.2",
     "styled-components": "^5.3.5",
     "typescript": "^4.6.3"
+  },
+  "scripts": {
+    "docusaurus": "docusaurus",
+    "start": "docusaurus start",
+    "build": "docusaurus build --out-dir build/open-source/spectacle",
+    "clear": "docusaurus clear",
+    "serve": "formideploy serve",
+    "deploy:stage": "formideploy deploy --staging",
+    "deploy:prod": "formideploy deploy --production",
+    "write-translations": "docusaurus write-translations",
+    "write-heading-ids": "docusaurus write-heading-ids",
+    "typecheck": "tsc",
+    "lint": "wireit",
+    "lint:fix": "wireit",
+    "prettier": "wireit",
+    "prettier:fix": "wireit"
+  },
+  "wireit": {
+    "lint": {
+      "command": "nps lint:pkg -- -- \"*.js\" src",
+      "files": [
+        "../../.eslintignore",
+        "../../.eslintrc",
+        "*.js",
+        "src/**"
+      ],
+      "output": [],
+      "packageLocks": [
+        "pnpm-lock.yaml"
+      ]
+    },
+    "lint:fix": {
+      "command": "pnpm run lint || nps lint:pkg:fix -- -- \"*.js\" src",
+      "files": [
+        "../.eslintignore",
+        "../.eslintrc",
+        "*.js",
+        "src/**"
+      ],
+      "output": [],
+      "packageLocks": [
+        "pnpm-lock.yaml"
+      ]
+    },
+    "prettier": {
+      "command": "nps prettier:website -- -- \"*.js\" src",
+      "files": [
+        "../.prettierignore",
+        "../.prettierrc",
+        "*.js",
+        "src/**"
+      ],
+      "output": [],
+      "packageLocks": [
+        "pnpm-lock.yaml"
+      ]
+    },
+    "prettier:fix": {
+      "command": "pnpm run prettier || nps prettier:website:fix -- -- \"*.js\" src",
+      "files": [
+        "../.prettierignore",
+        "../.prettierrc",
+        "*.js",
+        "src/**"
+      ],
+      "output": [],
+      "packageLocks": [
+        "pnpm-lock.yaml"
+      ]
+    }
   },
   "browserslist": {
     "production": [

--- a/website/package.json
+++ b/website/package.json
@@ -73,7 +73,7 @@
       ]
     },
     "prettier": {
-      "command": "nps prettier:website -- -- \"*.js\" src",
+      "command": "nps prettier:website -- -- src",
       "files": [
         "../.prettierignore",
         "../.prettierrc",

--- a/website/src/components/global/body-copy.tsx
+++ b/website/src/components/global/body-copy.tsx
@@ -10,11 +10,7 @@ type BodyCopy = {
 
 const BodyCopy = ({ className, theme, children }: BodyCopy) => {
   const isLight = theme === 'Light' || theme === 'Color';
-  const classNames = clsx(
-    styles.bodyCopy,
-    isLight && styles.light,
-    className
-  );
+  const classNames = clsx(styles.bodyCopy, isLight && styles.light, className);
 
   return <p className={classNames}>{children}</p>;
 };

--- a/website/src/utils/numbers.js
+++ b/website/src/utils/numbers.js
@@ -42,4 +42,4 @@ module.exports.romanize = function romanize(num) {
   let i = 3;
   while (i--) roman = (key[+digits.pop() + i * 10] || '') + roman;
   return Array(+digits.join('') + 1).join('M') + roman;
-}
+};

